core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/ExtensionLoader.java

```diff
@@ -92,8 +93,20 @@ public static void loadVault(ServiceExtensionContext context) {
     }
 
     public static @NotNull Telemetry loadTelemetry() {
-        // TODO: enable overriding telemetry through an extension
-        return new Telemetry(GlobalOpenTelemetry.get());
+        return new Telemetry(loadOpenTelemetry());
+    }
+
+    public static @NotNull OpenTelemetry loadOpenTelemetry() {
+        var loader = ServiceLoader.load(OpenTelemetry.class);
+        var openTelemetries = loader.stream().map(ServiceLoader.Provider::get).collect(Collectors.toList());
+        return loadOpenTelemetry(openTelemetries);
+    }
+
+    static @NotNull OpenTelemetry loadOpenTelemetry(List<OpenTelemetry> openTelemetries) {
+        if (openTelemetries.size() > 1) throw new IllegalStateException(
+                String.format("Found %s OpenTelemetry implementations. " +
+                        "Please provide only one OpenTelemetry service provider.", openTelemetries.size()));
```

Comment:
Is concatenation here intended (or added by IDE)?

Reply:
I did it on purpose because it easier to read as it is a long line. But I can remove it, are you aware of a convention about this in the EDC repo?

Reply:
If Checkstyle doesn't complain then it's aligned with EDC guidelines.  Feel free to leave it as you prefer, just thought it was IDE autoformat :)